### DISCUSSION TITLE
get data from remote (shared) account

### DISCUSSION THREAD

**Discussion by dbd-infra** [gruntwork-io/knowledge-base#693] [How do I]
We store secret data in `aws_ssm_paramaters` under shared account. 
We created a role which allows to assume it from dev account (Trust reletionships in shared acc) and want to get data from it like:
```
provider "aws" {
  alias   = "shared"
  region  = "us-east-1"
  assume_role {
    role_arn = format("arn:aws:iam::%s:role/assume_from_remote", var.shared_account_id)
	session_name = "terraform"
  }
}

data "aws_ssm_parameter" "some-api-key" {
  provider = aws.shared
  name     = "/config/${var.account_name}/SOME_API_KEY"
}
```  
The next step is to allow dev account to assume role in shared (in dev acc).  But it's not clear how to configure it.

---

<ins datetime="2023-04-05T15:59:15Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110057">Tracked in ticket #110057</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
I was thinking to extract `data "aws_ssm_parameter" ` to a separate dependency where in HCL file can be declared `iam_role` with the right values

https://terragrunt.gruntwork.io/docs/features/work-with-multiple-aws-accounts/
